Sumologic

scottdixon opened the pull request:
Users can now `-j https://sumo/collection/endpoint` to send the JSON results to sumo.

`Q.fcall` wasn't behaving with `logs-to-sumologic`, manually creating a promise.

Results include: 
`startupTime` - Session start time
`time` - Entire time of test, including session start time
`stack` - If this is null, the test was successful.

@jlipps please review. Also invited you to the demo Sumo account.

jlipps commented on src/utils.es6.js:
```diff
@@ -0,0 +1,15 @@
+import Sumologic from 'logs-to-sumologic';
+
+export function sendToSumo (url, logs) {
+  const sumologic = Sumologic.createClient({
```
`Sumologic.createClient({url});` is a bit shorter

jlipps commented on src/utils.es6.js:
```diff
@@ -0,0 +1,15 @@
+  return new Promise(function(resolve, reject) {
+    sumologic.log(logs, function (err) {
+      if (err) {
+        reject(err);
```
I guess we need a `return` here

jlipps commented:
Looking good! Do you recall offhand whether `results` also includes the testType? We probably want that so we can appropriately relativize the time data (different tests take different amounts of time). Do you think it could also be interesting to send in the runsauce version from package.json? Because the same test type might be different in different versions of runsauce.

scottdixon commented:
Yep, we get something like: `name: 'Test of safariIgnoreFraudWarning'`

Good call on the package.json version. Added!

Also, added key 'status' in lieu of 'error'. This gives us a value of `success` or `failure`. The entire stack trace was wreaking sumo havoc. Do you think success/failure is sufficient for our purposes, @jlipps? It means we don't see the reason for the error but I don't think it's relevant for our "works or not" testing here.

scottdixon commented:
Save you the login (two successes and a failure):
![image](https://cloud.githubusercontent.com/assets/4726068/16057563/83fc8b32-322f-11e6-843c-0bd0325f5d8b.png)

jlipps commented on src/runner.es6.js:
```diff
@@ -508,5 +530,5 @@ export async function run (opts, log = true, statusFn = null) {
     statusFn({localServer: 'stopped'});
   }
-  return reportSuite(results, Date.now() - start, log);
+  return reportSuite(results, Date.now() - start, log, opts);
```
for clarity, might want to do `return await reportSuite(...)` here

jlipps commented:
one last comment then good to merge!
